edb/server/compiler/sertypes.py

```diff
@@ -205,7 +206,17 @@ def _describe_type(self, t, view_shapes, view_shapes_metadata,
         elif isinstance(t, s_types.Collection):
             raise errors.SchemaError(f'unsupported collection type {t!r}')
 
-        elif view_shapes.get(t):
+        elif isinstance(t, s_objtypes.ObjectType):
+            if t not in view_shapes:
+                # If we have an object type that isn't attached to a view,
+                # it must be the case that it doesn't actually appear in
+                # the output, so just return *something* that is well
+                # formed.
+                return self._describe_type(
+                    self.schema.get('std::uuid'), view_shapes,
```

Comment:
I feel like a better fix would be to pretend that a `view_shape` exists, but is empty, or, even better, consisting entirely of implicit pointers.  Otherwise there is a danger of the frontend getting confused by an object not being represented properly.